docs/Contributing/Testing-and-local-development.md

```diff
@@ -448,3 +449,123 @@ MinIO also offers a web interface at http://localhost:9001. Credentials are `min
+
+Users will be able to generate this using `fleetctl generate mdm-apple` in the near future, but for now, ask in [#g-mdm](https://fleetdm.slack.com/archives/C03C41L5YEL) how to get them.
+
+### SCEP setup
+
+Another key pair is used for [SCEP](https://github.com/fleetdm/fleet/blob/main/tools/mdm/apple/glossary-and-protocols.md#scep-simple-certificate-enrollment-protocol), in the near future, this will also be generated by `fleetctl generate mdm-apple`, but for now please use this:
+
+```
+openssl genrsa -out fleet-mdm-apple-scep.key 4096
+
+openssl req -x509 -new -nodes -key fleet-mdm-apple-scep.key -sha256 -days 1826 -out fleet-mdm-apple-scep.crt -subj '/CN=Fleet Root CA/C=US/O=Fleet DM.'
+```
+
+### Running the server
+
+Try to store all the certificates and tokens you generated in the earlier steps together in a safe place outside of the repo, then start the server with:
+
+```
+FLEET_MDM_APPLE_ENABLE=1 \
+FLEET_MDM_APPLE_SCEP_CHALLENGE=scepchallenge \
+FLEET_MDM_APPLE_SCEP_CERT=/path/to/fleet-mdm-apple-scep.crt \
+FLEET_MDM_APPLE_SCEP_KEY=/path/to/fleet-mdm-apple-scep.key \
+FLEET_MDM_APPLE_BM_SERVER_TOKEN=/path/to/dep_encrypted_token.p7m \
+FLEET_MDM_APPLE_BM_CERT=/path/to/fleet-apple-mdm-bm-public-key.crt \
+FLEET_MDM_APPLE_BM_KEY=/path/to/fleet-apple-mdm-bm-private.key \
+FLEET_MDM_APPLE_APNS_CERT=/path/to/mdmcert.download.push.pem \
+FLEET_MDM_APPLE_APNS_KEY=/path/to/mdmcert.download.push.key \
+ ./build/fleet serve --dev --dev_license --logging_debug
+```
+
+The server needs to run behind TLS with a valid certificate, so in a separate terminal window/tab, create a local tunnel to your server using `ngrok` (`brew install ngrok/ngrok/ngrok` if you don't have it.)
+
+```
+ngrok http https://localhost:8080
+```
+
+Don't forget to edit your Fleet server settings (through the UI or `fleetctl`) to use the URL `ngrok` provides to you. You need to do this whenever you restart `ngrok`.
+
+### Testing MDM
+
+To test MDM, you'll need one or more virtual machines (VMs) that you can use to enroll to your server.
+
+Choose and download a VM software, some options:
+
+- VMware Fusion: https://www.vmware.com/products/fusion.html
+- UTM: https://mac.getutm.app/
+
+If you need a license ask @lukeheath or @zwass in [#g-mdm](https://fleetdm.slack.com/archives/C03C41L5YEL)
+
+With the software in place, you need to create a VM and install macOS, the steps to do this vary depending on your software of choice.
+
+We've used [this guide](https://travellingtechguy.blog/vmware-dep/) in the past, please reach out in [#g-mdm](https://fleetdm.slack.com/archives/C03C41L5YEL) before starting so you can get the right serial numbers.
+
+#### Testing manual enrollment
+
+1. Create a manual profile with:
+
+```
+fleetctl apple-mdm enrollment-profiles create-manual
```

Comment:
I think this will soon change with the manual enrollment support in fleet desktop, but that can be updated when we have the full feature rolled out.

Reply:
we're moving fast!